File: javascript/users/lookup.js
/**
 * User Lookup - X API v2
 * 
 * Endpoint: GET https://api.x.com/2/users/by
 * Docs: https://developer.x.com/en/docs/twitter-api/users/lookup/api-reference/get-users-by
 * 
 * Authentication: Bearer Token (App-only) or OAuth (User Context)
 * Required env vars: BEARER_TOKEN
 */

const needle = require('needle');

const token = process.env.BEARER_TOKEN;
const endpointURL = "https://api.x.com/2/users/by";

async function getRequest() {
    // Usernames to look up (up to 100 comma-separated)
    const params = {
        "usernames": "XDevelopers,X",
        "user.fields": "created_at,description,public_metrics"
    };

    const res = await needle('get', endpointURL, params, {
        headers: {
            "User-Agent": "v2UserLookupJS",
            "authorization": `Bearer ${token}`
        }
    });

    if (res.body) {
        return res.body;
    } else {
        throw new Error('Unsuccessful request');
    }
}

(async () => {
    try {
        const response = await getRequest();
        console.dir(response, { depth: null });
    } catch (e) {
        console.log(e);
        process.exit(-1);
    }
    process.exit();
})();

File: ruby/users/lookup.rb
# User Lookup - X API v2
#
# Endpoint: GET https://api.x.com/2/users/by
# Docs: https://developer.x.com/en/docs/twitter-api/users/lookup/api-reference/get-users-by
#
# Authentication: Bearer Token (App-only) or OAuth (User Context)
# Required env vars: BEARER_TOKEN

require 'json'
require 'typhoeus'

bearer_token = ENV["BEARER_TOKEN"]

url = "https://api.x.com/2/users/by"

# Usernames to look up (up to 100 comma-separated)
query_params = {
  "usernames": "XDevelopers,X",
  "user.fields": "created_at,description,public_metrics"
}

def lookup_users(url, bearer_token, query_params)
  options = {
    method: 'get',
    headers: {
      "User-Agent": "v2UserLookupRuby",
      "Authorization": "Bearer #{bearer_token}"
    },
    params: query_params
  }

  request = Typhoeus::Request.new(url, options)
  response = request.run

  return response
end

response = lookup_users(url, bearer_token, query_params)
puts response.code, JSON.pretty_generate(JSON.parse(response.body))
